### PR TITLE
[release-1.23.1] Update helm chart version to v0.23.1

### PR DESCRIPTION
**Pull request by JaneLiuL**
bump helm chart version to v0.23.1

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
/ok-to-test
this is a bit confusing, but since we're technically only making a change to the Helm chart deployments all we really need to update is just the `version` in `Charts.yaml` to `23.1`. The Descheduler itself can still just be `v0.23.0` since we didn't make any functional changes to the application itself

File: charts/descheduler/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: descheduler
-version: 0.23.0
-appVersion: 0.23.0
+version: 0.23.1
+appVersion: 0.23.1
```
[the remark this answers is not in the document]
Comment (JaneLiuL) [reply charts/descheduler/Chart.yaml:4]: thanks, i will update now~

[the remark this answers is not in the document]
Comment (JaneLiuL) [reply charts/descheduler/Chart.yaml]: fix done :)

File: docs/user-guide.md [hunk 2]
```diff
@@ -2,14 +2,16 @@
...
-------------------- |-----------------------------------------------------|-------------------------|
-v0.22.0             | k8s.gcr.io/descheduler/descheduler:v0.22.0          | AMD64<br>ARM64<br>ARMv7 |
-v0.21.0             | k8s.gcr.io/descheduler/descheduler:v0.21.0          | AMD64<br>ARM64<br>ARMv7 |
-v0.20.0             | k8s.gcr.io/descheduler/descheduler:v0.20.0          | AMD64<br>ARM64          |
-v0.19.0             | k8s.gcr.io/descheduler/descheduler:v0.19.0          | AMD64                   |
-v0.18.0             | k8s.gcr.io/descheduler/descheduler:v0.18.0          | AMD64                   |
-v0.10.0             | k8s.gcr.io/descheduler/descheduler:v0.10.0          | AMD64                   |
+Descheduler Version | Container Image                            | Architectures           |
+------------------- |--------------------------------------------|-------------------------|
+v0.23.1             | k8s.gcr.io/descheduler/descheduler:v0.23.1 | AMD64<br>ARM64<br>ARMv7 |
```
Comment (damemi) [inline docs/user-guide.md:7]: Please update Descheduler version and Container Image to `v0.23.0`

**Comment by damemi** [comment]
one nit, looks good otherwise
/approve

File: docs/user-guide.md (hunk 2 above)
Comment (JaneLiuL) [reply docs/user-guide.md]: yes. done :) sorry for my mistake

**Pull request merged**